@shared/api/internal.ts

```diff
@@ -280,14 +279,32 @@ export const getAccountBalances = async ({
     networkPassphrase,
   });
 
-  let balances = null;
+  let balances: any = null;
   let isFunded = null;
   let subentryCount = 0;
 
   try {
-    ({ balances, subentryCount } = await dataProvider.fetchAccountDetails());
-    // let's filter out liquidity pool shares until freighter and wallet-sdk support
-    balances = omitBy(balances, (b: any) => b.liquidity_pool_id) as Balances;
+    const resp = await dataProvider.fetchAccountDetails();
+    balances = resp.balances;
+    subentryCount = resp.subentryCount;
+
+    for (let i = 0; i < Object.keys(resp.balances).length; i++) {
```

Comment:
til (after much befuddlement) `.forEach` doesn't wait on the `await` calls inside its loop 🤷

Reply:
Argh, yeah. Sorry, one of those annoying gotchas. Glad you found the solution!